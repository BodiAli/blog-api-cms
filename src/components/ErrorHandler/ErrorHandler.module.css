.error {
  margin-top: 4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  font-family: "roboto", Arial, sans-serif;
}

.error > h2 {
  font-size: 2rem;
}

.error > p {
  color: crimson;
}

.error > button {
  background-color: rgb(72, 130, 180);
  border: none;
  border-radius: 5px;
  color: white;
  padding: 0.4rem 0.8rem;
  font-size: 1rem;
  cursor: pointer;
  transition: background-color 200ms;
  &:hover {
    background-color: rgb(102, 148, 188);
  }
}
