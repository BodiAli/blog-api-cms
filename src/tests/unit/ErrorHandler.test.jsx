import { render, screen } from "@testing-library/react";
import { createRoutesStub } from "react-router";
import userEvent from "@testing-library/user-event";
import ErrorHandler from "../../components/ErrorHandler/ErrorHandler";

function ThrowErrorComponent() {
  throw new Error("ERROR!!!");
}

describe("ErrorHandler component", () => {
  test("Should render ERROR!!! message with a reload button", () => {
    const Stub = createRoutesStub([
      {
        path: "/",
        Component: ThrowErrorComponent,
        ErrorBoundary: ErrorHandler,
      },
    ]);
    const { container } = render(<Stub initialEntries={["/"]} />);

    expect(container).toMatchSnapshot();
  });

  test("Should call reload page when reload button is clicked", async () => {
    const user = userEvent.setup();

    window.location = { reload: vi.fn() };

    const Stub = createRoutesStub([
      {
        path: "/",
        Component: ThrowErrorComponent,
        ErrorBoundary: ErrorHandler,
      },
    ]);

    render(<Stub initialEntries={["/"]} />);

    await user.click(screen.getByRole("button"));

    expect(window.location.reload).toHaveBeenCalledOnce();
  });
});
