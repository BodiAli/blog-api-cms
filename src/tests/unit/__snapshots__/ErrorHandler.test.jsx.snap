// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorHandler component > Should render ERROR!!! message with a reload button 1`] = `
<div>
  <div
    class="_error_2449a6"
  >
    <h2>
      Unexpected error occurred!
    </h2>
    <p>
      ERROR!!!
    </p>
    <button>
      Reload page
    </button>
  </div>
</div>
`;
